docs/components.md

```diff
@@ -148,65 +197,79 @@ class AnotherES6ClassComponent {
 
-Functionally minded developers may prefer using the "closure component" syntax:
+If a state change occurs that is not as a result of any of the above conditions (e.g. after a `setTimeout`), then you can use `m.redraw()` to trigger a redraw manually.
+
+#### Closure Component State
+
+With a closure component state can simply be maintained by variables that are declared within the outer function. For example:
 
 ```javascript
-function closureComponent(vnode) {
-	// vnode.state is undefined at this point
-	var kind = "closure component"
+function ComponentWithState() {
+	// Variables that hold component state
+	var count = 0
 
 	return {
 		view: function() {
-			return m("div", "Hello from a " + kind)
-		},
-		oncreate: function() {
-			console.log("We've created a " + kind)
+			return m("div",
+				m("p", "Count: " + count),
+				m("button", {
+					onclick: function() {
+						count += 1
+					}
+				}, "Increment count")
+			)
 		}
 	}
 }
 ```
 
-The returned object must hold a `view` function, used to get the tree to render.
-
-They can be consumed in the same way regular components can.
+Any functions declared within the closure also have access to its state variables.
 
 ```javascript
-// EXAMPLE: via m.render
-m.render(document.body, m(closureComponent))
+function ComponentWithState() {
+	var count = 0
 
-// EXAMPLE: via m.mount
-m.mount(document.body, closureComponent)
+	function increment() {
+		count += 1
+	}
 
-// EXAMPLE: via m.route
-m.route(document.body, "/", {
-	"/": closureComponent
-})
+	function decrement() {
+		count -= 1
+	}
 
-// EXAMPLE: component composition
-function anotherClosureComponent() {
 	return {
 		view: function() {
-			return m("main", [
-				m(closureComponent)
-			])
+			return m("div",
+				m("p", "Count: " + count),
+				m("button", {
+					onclick: increment
+				}, "Increment"),
+				m("button", {
+					onclick: decrement
+				}, "Decrement")
+			)
 		}
 	}
 }
 ```
 
-#### Mixing component kinds
+A big advantage of closure components is that we don't need to worry about binding `this` when attaching event handler callbacks. In fact `this` is never used at all and we never have to think about `this` context ambiguities.
 
-Components can be freely mixed. A Class component can have closure or POJO components as children, etc...
+#### POJO and Class Component State
 
----
-
-### State
-
-Like all virtual DOM nodes, component vnodes can have state. Component state is useful for supporting object-oriented architectures, for encapsulation and for separation of concerns.
-
-The state of a component can be accessed three ways: as a blueprint at initialization, via `vnode.state` and via the `this` keyword in component methods.
+For POJO or class components the state of a component can be accessed three ways: as a blueprint at initialization, via `vnode.state` and via the `this` keyword in component methods.
```

Comment:
Few nits here:

- Could you split this section back out into class components then POJO components?
- Could you explain what you mean by "as a blueprint at initialization"? I'm having trouble making sense of what you're referring to.
- There's a giant code sample for closure component state, but there's no corresponding example for class or POJO components.

Reply:
"as a blueprint at initialization" is from the original text. I'm not exactly sure what it means either.

Reply:
@spacejack Yeah, let's just kill the language and replace it with something that makes more sense.

Reply:
Re-reading I get what it means. The following section describes "At initialization", and how you can initialize state. Eg:

```javascript
var ComponentWithInitialState = {
    data: "Initial content",
    view: function() {}
}
```

It does not cover the footgun of initializing with a non-scalar (eg object, array) property.

This section probably needs expansion along with examples as you suggested.

Reply:
That's of course if you choose to do that now, rather than wait for it to be merged. Also, if you do, could you link me or at least @mention me in the bug so I can follow the discussion?

Reply:
Let's merge this bit of work and I'll tackle it afterwards.